### ISSUE TITLE
Add operator overloads

### ISSUE THREAD

**Issue by AlexanderJDupree** [AlexanderJDupree/LinkedListsCPP#7]
- [x] overload = operator
- [x] overload equality and comparison operators
~~- overload [] operator for element access~~ See comments for justification

**Comment by AlexanderJDupree** [comment]
Equality operator implemented in PR #29

**Comment by AlexanderJDupree** [comment]
I'll be getting to work on the assignment and element access operators for my next commit

**Comment by AlexanderJDupree** [comment]
@jake-bickle, How do you think we should implement the [ ] operator for element access? 

Preferably, the client could use the operator like this:
```c++
LinkedList<int> list { 1, 2, 3, 4, 5};

std::cout << list[2] << std::endl;

// Prints 3 to the console
```
However, implementing this is difficult, as the nature of the linked list prevents random access. So if we attempted to loop through the list using index access we would square the time complexity of the algorithm. 

For example, here is the naive implementation I was thinking of:

```c++
template <typename T>
T& LinkedList<T>::operator[] (size_type n)
{
    iterator it = begin();
    while (n >= 0)
    {
        ++it;
        --n;
    }
    return *it;
}
```

Ignoring, the other faults of this implementation (like what happens when 'n' is out of range), because I can only start at the beginning of the list, each time the [ ] operator is called we have to travel through the entirety of the list.

**Comment by jake-bickle** [comment]
@AlexanderJDupree You're right, the only element we have access to is the head and the tail (which isn't of much use here). To get the nth element, it would be n time to locate it.

One idea I can think of is attaching a number to each node to keep track of its place in the list. However, this basically turns the list into a tricky array and comes with its own set of inefficiencies ( Making sure every node is in order after modifying the list ).

However, is this necessary? Sure it's possible to access the list in such a fashion (just like it's theoretically possible to use subtraction arithmetic on a list iterator) but is it needed? Neither the standard forward_list or list use [] operators, and its "unnatural" to access a list this way. If the user needs that functionality, they'd best use a different data structure.

**Comment by AlexanderJDupree** [comment]
Yeah that's pretty much what I'm thinking too. The only way to implement this properly would be to create some sort of cursor member that keeps track of the last element it accessed and associate it to an index in the list. However, implementing a cursor like this would create more headaches because now we would need to initialize the cursor, track the cursor, make sure we don't delete what the cursor is pointing to etc. etc. 

I think we should scrap the [ ] operator for now, and use iterators as the primary method of traversing our list.

**Comment by jake-bickle** [comment]
I agree. As I see it now, iterators should be the only means of accessing the elements within the list.

**Comment by AlexanderJDupree** [comment]
copy assignment operator implemented in PR #40
